src/main/php/web/Request.class.php

```diff
@@ -17,16 +17,28 @@ class Request {
   private $cookies= null;
   private $method, $uri, $input;
 
-  /** @param web.io.Input $input */
-  public function __construct(Input $input) {
-    foreach ($input->headers() as $name => $value) {
-      $this->headers[$name]= $value;
-      $this->lookup[strtolower($name)]= $name;
+  /** @param web.io.Input|self $arg */
+  public function __construct($arg) {
+    if ($arg instanceof self) {
```

Comment:
The `Request` object is created for every request, maybe we can find a solution without introducing an extra `if`?